routing-configurer/src/test/java/no/vegvesen/ixn/federation/qpid/QpidClientIT.java

```diff
@@ -154,12 +154,73 @@ public void addOneConsumeRuleBeforeLastRule() {
 		acl.add("ACL ALLOW-LOG " + SERVICE_PROVIDERS_GROUP_NAME + " ACCESS VIRTUALHOST name = \"localhost\"");
 		acl.add("ACL ALLOW-LOG interchange CONSUME QUEUE name = \"onramp\"");
 		acl.add("ACL DENY-LOG ALL ALL ALL");
-		LinkedList<String> newAcl = new LinkedList<>(client.addOneConsumeRuleBeforeLastRule("king_harald", "king_harald", acl));
+		String newAclEntry = String.format("ACL ALLOW-LOG king_harald CONSUME QUEUE name = \"king_harald\"");
+		LinkedList<String> newAcl = new LinkedList<>(client.addOneConsumeRuleBeforeLastRule(acl, newAclEntry));
 		assertThat(acl.getFirst()).isEqualTo(newAcl.getFirst());
 		assertThat(acl.getLast()).isEqualTo(newAcl.getLast());
 		assertThat(newAcl.get(newAcl.size() - 2)).contains("king_harald");
 	}
 
+	@Test
+	public void readAccessIsAdded() {
+		String subscriberName = "king_harald";
+		String queueName = "king_harald";
+
+		client.addReadAccess(subscriberName, queueName);
+
+		List<String> newAclRules = client.getACL();
+
+		String newAclEntry = String.format("ACL ALLOW-LOG king_harald CONSUME QUEUE name = \"king_harald\"");
+
+		assertThat(newAclRules.get(newAclRules.size() - 2)).isEqualTo(newAclEntry);
+	}
+
+	@Test
+	public void readAccessIsRemoved() {
+		String subscriberName = "king_harald";
+		String queueName = "king_harald";
+
+		client.removeReadAccess(subscriberName, queueName);
+
+		List<String> newAclRules = client.getACL();
+
+		String deletedAclEntry = String.format("ACL ALLOW-LOG king_harald CONSUME QUEUE name = \"king_harald\"");
+
+		assertThat(newAclRules).doesNotContain(deletedAclEntry);
+	}
+
+	@Test
+	public void writeAccessIsAdded() {
+		String subscriberName = "king_harald";
+		String queueName = "king_harald";
+
+		client.addWriteAccess(subscriberName, queueName);
+
+		List<String> newAclRules = client.getACL();
+
+		//String newAclEntry = String.format("ACL ALLOW-LOG king_harald PUBLISH EXCHANGE name = \"\" routingkey = \"king_harald\"");
+
+		assertThat(newAclRules.get(newAclRules.size() - 2)).startsWith("ACL ALLOW-LOG king_harald PUBLISH EXCHANGE");
+		assertThat(newAclRules.get(newAclRules.size() - 2)).contains("name = \"\"");
+		assertThat(newAclRules.get(newAclRules.size() - 2)).contains("routingkey = \"king_harald\"");
+	}
+
+	@Test
+	public void writeAccessIsRemoved() {
+		String subscriberName = "king_harald";
+		String queueName = "king_harald";
+
+		client.removeWriteAccess(subscriberName, queueName);
+
+		List<String> newAclRules = client.getACL();
+
+		String deletedAclEntry = String.format("ACL ALLOW-LOG king_harald PUBLISH EXCHANGE name = \"\" routingkey = \"king_harald\"");
+		String deletedAclEntry1 = String.format("ACL ALLOW-LOG king_harald PUBLISH EXCHANGE routingkey = \"king_harald\" name = \"\"");
+
+		assertThat(newAclRules).doesNotContain(deletedAclEntry);
```

Comment:
This can give us false positives, if we remove one of the lines from the set, but switches the order of the attributes in the string. I believe in this case, you would actually spin through the set and make sure none of the strings matches the three calls in the previous test (startswith, containes, contains).

Reply:
ok